bayes_search.py

```diff
@@ -0,0 +1,441 @@
+import numpy as np
+import numpy.typing as npt
+
+from typing import List, Tuple, Optional, Union
+
+from .config.cfg import SweepConfig
+from .run import SweepRun, RunState
+from .params import HyperParameter, HyperParameterSet
+from sklearn import gaussian_process as sklearn_gaussian
+from scipy import stats as scipy_stats
+
+from ._types import floating, integer
+
+NUGGET = 1e-10
+
+
+def fit_normalized_gaussian_process(
+    X: npt.ArrayLike, y: npt.ArrayLike, nu: floating = 1.5
+) -> Tuple[sklearn_gaussian.GaussianProcessRegressor, floating, floating]:
+    gp = sklearn_gaussian.GaussianProcessRegressor(
+        kernel=sklearn_gaussian.kernels.Matern(nu=nu),
+        n_restarts_optimizer=2,
+        alpha=0.0000001,
+        random_state=2,
+    )
+    if len(y) == 1:
+        y = np.array(y)
+        y_mean = y[0]
+        y_stddev = 1
+    else:
+        y_mean = np.mean(y)
+        y_stddev = np.std(y) + 0.0001
+    y_norm = (y - y_mean) / y_stddev
+    gp.fit(X, y_norm)
+    return gp, y_mean, y_stddev
+
+
+def sigmoid(x: npt.ArrayLike) -> npt.ArrayLike:
+    return np.exp(-np.logaddexp(0, -x))
+
+
+def random_sample(X_bounds: npt.ArrayLike, num_test_samples: integer) -> npt.ArrayLike:
+    num_hyperparameters = len(X_bounds)
+    test_X = np.empty((num_test_samples, num_hyperparameters))
+    for ii in range(num_test_samples):
+        for jj in range(num_hyperparameters):
+            if type(X_bounds[jj][0]) == int:
+                assert type(X_bounds[jj][1]) == int
+                test_X[ii, jj] = np.random.randint(X_bounds[jj][0], X_bounds[jj][1])
+            else:
+                test_X[ii, jj] = (
+                    np.random.uniform() * (X_bounds[jj][1] - X_bounds[jj][0])
+                    + X_bounds[jj][0]
+                )
+    return test_X
+
+
+def predict(
+    X: npt.ArrayLike, y: npt.ArrayLike, test_X: npt.ArrayLike, nu: floating = 1.5
+) -> Tuple[npt.ArrayLike, npt.ArrayLike]:
+    gp, norm_mean, norm_stddev = fit_normalized_gaussian_process(X, y, nu=nu)
+    y_pred, y_std = gp.predict([test_X], return_std=True)
+    y_std_norm = y_std * norm_stddev
+    y_pred_norm = (y_pred * norm_stddev) + norm_mean
+    return y_pred_norm[0], y_std_norm[0]
+
+
+def train_runtime_model(
```

Comment:
Not used any more, it looks like the unused code was removed from next_sample

Reply:
fixed